src/ladds/Simulation.cpp

```diff
@@ -301,34 +387,63 @@ size_t Simulation::simulationLoop(AutoPas_t &autopas,
       // convert timer value from ns to s
       const size_t secondsSinceStart = timers.total.getTotalTime() / static_cast<size_t>(1e9);
       if (secondsSinceStart > timeout) {
+        // FIXME: MPI SYNC THIS?
         SPDLOG_LOGGER_INFO(logger.get(),
                            "Simulation timeout hit! Time since simulation start ({} s) > Timeout ({} s))",
                            secondsSinceStart,
                            timeout);
         // set the config to the number of completed iterations (hence no +/-1) for the timer calculations.
         config.setValue("sim/iterations", iteration);
         // abort the loop by increasing the loop counter. This also leads to triggering the visualization
-        iteration = iterations - 1;
+        iteration = lastIteration;
       }
     }
 
     timers.output.start();
-    if (iteration % progressOutputFrequency == 0 or iteration == (iterations - 1)) {
+    if (iteration % progressOutputFrequency == 0 or iteration == lastIteration) {
```

Comment:
this function is getting very long again, maybe break it up?

Reply:
I extracted some output. Feel free to make further suggestions